src/Phinx/Migration/Manager.php

```diff
@@ -173,12 +181,36 @@ public function migrate($environment, $version = null)
                 break;
             }
 
-            if (!in_array($migration->getVersion(), $versions)) {
+            if (in_array($migration->getVersion(), $pendingVersions)) {
+                list($versions, $pendingVersions) = $this->waitOnPendingMigration($migration, $env);
+            }
+
+            if (! in_array($migration->getVersion(), $versions)) {
                 $this->executeMigration($environment, $migration, MigrationInterface::UP);
             }
         }
     }
-    
+
+    protected function waitOnPendingMigration(MigrationInterface $migration, Environment $env)
+    {
+        $timeWaited = 0;
+        $this->getOutput()->writeln('');
+        do {
+            if ($timeWaited % (static::POLL_DELAY * (int) static::POLL_MESSAGE_FREQUENCY) == 0) {
```

Comment:
casting `POLL_MESSAGE_FREQUENCY` to int ensures that the result of `(POLL_DELAY * (int) POLL_MESSAGE_FREQUENCY)` is a multiple of `POLL_DELAY`.

`$timeWaited` is also a multiple of `POLL_DELAY`, so the `int` ensures that we always write to the screen when expected